mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2056,7 +2056,8 @@ where
     };
     let mut history_map: HashMap<H256Json, TransactionDetails> = history
         .into_iter()
-        .map(|tx| (H256Json::from(tx.tx_hash.as_bytes()), tx))
+        .filter(|tx| H256Json::from_str(&tx.tx_hash).is_ok())
+        .map(|tx| (H256Json::from_str(&tx.tx_hash).unwrap(), tx))
```

Comment:
There was an unexpected bug here at the creation of the history_map. Using `H256Json::from(tx.tx_hash.as_bytes())` created a different hash than the hashes inside the `tx_ids` vector. As a result, existing transactions in the history_map were ignored and all the transactions in the `tx_ids` were inserted repeatedly into the `history_map`. Creating the H256 by `H256Json::from_str(&tx.tx_hash)` fixed the problem.

Reply:
Good catch :)